code/modules/projectiles/projectile.dm

```diff
@@ -119,23 +133,83 @@
 /obj/item/projectile/Process_Spacemove(var/movement_dir = 0)
 	return 1 //Bullets don't drift in space
 
-
-/obj/item/projectile/proc/fire()
-	set waitfor = 0
-	while(loc)
-		if(!paused)
-			if((!( current ) || loc == current))
-				current = locate(Clamp(x+xo,1,world.maxx),Clamp(y+yo,1,world.maxy),z)
-			step_towards(src, current)
-			if(original && (original.layer>=2.75) || ismob(original))
-				if(loc == get_turf(original))
-					if(!(original in permutated))
-						Bump(original, 1)
-			Range()
-		sleep(1)
+/obj/item/projectile/proc/fire(var/setAngle)
+	if(setAngle)
+		Angle = setAngle
+	if(!legacy) //new projectiles
+		set waitfor = 0
+		while(loc)
+			if(!paused)
+				if((!( current ) || loc == current))
+					current = locate(Clamp(x+xo,1,world.maxx),Clamp(y+yo,1,world.maxy),z)
+
+				if(!Angle)
+					Angle=round(Get_Angle(src,current))
+				if(spread)
+					Angle += (rand() - 0.5) * spread
+				var/matrix/M = new
+				M.Turn(Angle)
+				transform = M
+
+				var/Pixel_x=round(sin(Angle)+16*sin(Angle)*2)
+				var/Pixel_y=round(cos(Angle)+16*cos(Angle)*2)
+				var/pixel_x_offset = pixel_x + Pixel_x
+				var/pixel_y_offset = pixel_y + Pixel_y

```

Comment:
You're going to name the var Pixel_x? Come on now. I shouldn't have to explain the perils of having an object var named pixel_x and a local var named Pixel_x.